src/app/space/create/deployments/deployments.component.spec.ts

```diff
@@ -40,69 +36,55 @@ class FakeDeploymentAppsComponent {
   @Input() applications: Observable<string[]>;
 }
 
+@Component({
+  template: '<alm-apps></alm-apps>'
+})
+class HostComponent { }
+
 describe('DeploymentsComponent', () => {
 
-  let component: DeploymentsComponent;
-  let fixture: ComponentFixture<DeploymentsComponent>;
-  let mockSvc: DeploymentsService;
-  let spaces: Spaces;
+  type Context = TestContext<DeploymentsComponent, HostComponent>;
+
+  let mockSvc: any = jasmine.createSpyObj('DeploymentsService', [
+    'getApplications', 'getEnvironments', 'getCpuStat', 'getMemoryStat'
+  ]);
+  let spaces = { current: Observable.of({ id: 'fake-spaceId' }) };
   let mockApplications = Observable.of(['foo-app', 'bar-app']);
   let mockEnvironments = Observable.of([
     { environmentId: 'a1', name: 'stage' },
     { environmentId: 'b2', name: 'prod' }
   ]);
 
-  beforeEach(() => {
-    mockSvc = {
-      getApplications: () => mockApplications,
-      getEnvironments: () => mockEnvironments,
-      getPodCount: () => { throw 'Not Implemented'; },
-      getVersion: () => { throw 'NotImplemented'; },
-      getCpuStat: (spaceId: string, envId: string) => Observable.of({ used: 1, total: 2 } as CpuStat),
-      getMemoryStat: (spaceId: string, envId: string) => Observable.of({ used: 1, total: 2 } as MemoryStat)
-    };
-
-    spaces = {
-      current: Observable.of({ id: 'fake-spaceId' })
-    } as Spaces;
-
-    spyOn(mockSvc, 'getApplications').and.callThrough();
-    spyOn(mockSvc, 'getEnvironments').and.callThrough();
-    spyOn(mockSvc, 'getPodCount').and.callThrough();
-    spyOn(mockSvc, 'getVersion').and.callThrough();
-    spyOn(mockSvc, 'getCpuStat').and.callThrough();
-    spyOn(mockSvc, 'getMemoryStat').and.callThrough();
-
-    TestBed.configureTestingModule({
-      imports: [ CollapseModule.forRoot() ],
-      declarations: [
-        DeploymentsComponent,
-        FakeDeploymentAppsComponent,
-        FakeDeploymentsResourceUsageComponent
-      ],
-      providers: [
-        { provide: DeploymentsService, useValue: mockSvc },
-        { provide: Spaces, useValue: spaces }
-      ]
-    });
-
-    fixture = TestBed.createComponent(DeploymentsComponent);
-    component = fixture.componentInstance;
+  beforeAll(() => {
+    mockSvc.getApplications.and.returnValue(mockApplications);
+    mockSvc.getEnvironments.and.returnValue(mockEnvironments);
+    mockSvc.getCpuStat.and.returnValue(Observable.of({ used: 1, total: 2 }));
+    mockSvc.getMemoryStat.and.returnValue(Observable.of({ used: 1, total: 2 }));
```

Comment:
Are CpuStat and MemoryStat used in any tests here? If not used, I'd rather they throw unimplemented. It follows with other locations that create this mock service. It then becomes clear that this component doesn't rely on those service functions.